scripts/librarian.mjs

```diff
@@ -0,0 +1,200 @@
+                }
+
+                libraryData.location = `${libraryDir}/${libraryData.location}`;
+
+                libraryData.customised = !!library.get('customised');
+                for (const copyrightHolder of library.get('copyrights')?.childNodes() || []) {
+                    if (copyrightHolder.type() !== 'element') {
+                        continue;
+                    }
+                    libraryData.copyrightHolders.push(copyrightHolder.text());
+                }
+
+                libraryList.push(libraryData);
+            }
+        }
+    }
+    return libraryList.sort((a, b) => a.location.localeCompare(b.location));
+};
+
+const generateLibraryJSON = async () => {
+    const startDir = process.cwd();
+    const outputFile = `${process.cwd()}/data/libraries.json`;
+    const libraries = await getLibraryData();
+    process.chdir(startDir);
+
+    writeFile(outputFile, JSON.stringify(libraries, null, 4));
+};
+
+const renderOutdatedNotice = (isOutdated) => {
+    if (!isOutdated) {
+        return '';
+    }
+
+    return `
+:::danger Outdated
+
+This library is not currently used in Moodle
+
+:::
+    `;
+};
+
+const renderCustomisedNotice = (isCustomised) => {
+    if (!isCustomised) {
+        return '';
+    }
+
+    return ' (with Moodle customisations)';
+};
+
+const renderCopyrightHolders = (copyrightHolders) => {
+    if (copyrightHolders.length === 0) {
+        return '';
+    }
+
+    return `- **Copyright holders**:
+${copyrightHolders.map((copyrightHolder) => `  - ${copyrightHolder}`).join(`
+`)}
+`;
+};
+
+const renderLibraryUrl = (libraryUrl) => {
+    if (libraryUrl.length === 0) {
+        return '';
+    }
+
+    return `- **URL**: [${libraryUrl}](${libraryUrl})`;
+};
+
+const renderLibraryData = (libraryData, isOutdated) => {
+    const libraryName = libraryData.name;
+    const libraryDescription = libraryData.description || '';
+    const libraryLocation = libraryData.location;
+    const libraryVersion = libraryData.version || '';
+    const libraryLicense = libraryData.license || '';
+    const libraryLicenseVersion = libraryData.licenseversion || '';
+    const libraryCustomised = libraryData.customised;
+    const libraryCopyrightHolders = libraryData.copyrightHolders || [];
+    const libraryUrl = libraryData.repository || '';
+
+    return `
+### ${libraryName}
+
+${libraryDescription}
+
+${renderOutdatedNotice(isOutdated)}
+- **Location**: ${libraryLocation}
+- **Version**: ${libraryVersion}${renderCustomisedNotice(libraryCustomised)}
+- **License**:  ${libraryLicense} ${libraryLicenseVersion}
+${renderLibraryUrl(libraryUrl)}
+${renderCopyrightHolders(libraryCopyrightHolders)}
+`.replaceAll(/ *$/gm, '');
+};
+
+const generateCredits = async () => {
+    const generateLibraryContentForFile = async (filePath, isOutdated) => JSON.parse(await readFile(filePath, 'utf-8'))
+        .sort((a, b) => a.location.localeCompare(b.location))
```

Comment:
I would suggest keeping the alphabetical order (by library name)... IMHO, it makes it easier to find them.

Reply:
In theory, sure...but right now we have some cases where there isn't a name.

`theme/boost/thirdpartylibs.xml` is missing a name for `amd/src/bootstrap/button.js` for example.

Until the xml is validated against an XSL I think we have to go with the one field which must be there (location) because other automatinos already rely on it within Moodle.

Reply:
Ohhhh, fair enough!